### PR TITLE
Stop Bundling GraphQL in the module

### PR DESCRIPTION
**Pull request by abhiaiyer91**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -156,5 +164,7 @@ function gql(/* arguments */) {
 gql.default = gql;
 gql.resetCaches = resetCaches;
 gql.disableFragmentWarnings = disableFragmentWarnings;
+gql.print = require('graphql/language/printer');
+gql.parser = parser
```
Comment (stubailo) [inline index.js]: It doesn't look like these two names are consistent - should it be "parse" rather than "parser"?

**Comment by stubailo** [comment]
Also it looks like this is adding a yarn lock file, I don't think we want that in the repository.

File: index.js (hunk 1 above)
Comment (abhiaiyer91) [inline index.js]: @stubailo I did a search:
https://github.com/apollographql/apollo-client/search?utf8=%E2%9C%93&q=parser

and saw some typings expecting `parser` 

What do you suggest? `parse`?

**Comment by abhiaiyer91** [comment]
@stubailo I added the lock file because I manage dependencies with yarn. Are we opposed to managing the dependencies and their versions with `yarn`?

**Comment by stubailo** [comment]
@abhiaiyer91 no, but you can install dependencies with yarn without adding a lock file. Lock files are perfect for applications, but not the right solution for packages, where we need to be constantly testing with new versions.

File: index.js (hunk 1 above)
Comment (stubailo) [inline index.js]: The module is called `parser`, but the function exported is called `parse`.

Comment (abhiaiyer91) [inline index.js]: Okay i will export the parse function instead of the whole parser

**Comment by abhiaiyer91** [comment]
Sounds good, will remove the yarn.lock

File: index.js (hunk 1 above)
Comment (abhiaiyer91) [inline index.js]: I was exporting the whole lib `var parser = require('graphql/language/parser');`

**Comment by abhiaiyer91** [comment]
Alright updates made!

File: index.js [hunk 2]
```diff
@@ -90,6 +93,11 @@ function stripLoc (doc, removeLocAtThisLevel) {
     delete doc.loc;
   }
 
+  if (doc.loc) {
```
Comment (jnwng) [inline index.js]: this could use a comment - im not fully certain why we'd want to delete these properties. linking to the issue should be sufficient

Comment (abhiaiyer91) [inline index.js]: Will add a comment, the parser's loc fields include references to the query string and as your AST is deeper it becomes more overhead to ship/store the query

**Comment by stubailo** [comment]
Yeah I think this is a great change although it is majorly breaking for anyone that was using `graphql-tag/parser` and `graphql-tag/printer`. So we should make sure to bump the major version IMO.

Also, are we worried about importing `print` unconditionally? It's possible that some apps might not be using that, maybe we should just instruct people to import from `graphql/language/printer` directly?

**Comment by stubailo** [comment]
(Consider that a sign-off, we can always fix it later if we think this increases the bundle size unnecessarily)

**Comment by abhiaiyer91** [comment]
@stubailo I think people should import printer/parser themselves. They are already going to have graphql due to the peer dependency.

**Comment by abhiaiyer91** [comment]
https://github.com/apollographql/graphql-tag/issues/49

**Comment by PowerKiKi** [comment]
Slightly off-topic but the lock file should be committed if you decide to go support yarn as is [well explained here](https://yarnpkg.com/blog/2016/11/24/lockfiles-for-all/).

Would you accept a PR for that ?

**Comment by jnwng** [comment]
thanks for sharing that article, i was fairly certain we _shouldn't_ be adding `yarn.lock` files. would be happy to accept a PR!

**Pull request merged**